### PR TITLE
Replaced use of post_live_migrate a new quark xenapi plugin

### PR DESCRIPTION
**Pull request by asadoughi**
Review/Merge #305 first, please.

### CODE REVIEW

**Comment by asadoughi** [comment]
I'd like to manually test building the rpm and invoking the plugin before removing the `WIP`.

**Comment by asadoughi** [comment]
I was able to build the RPM on Fedora VM in the cloud, but when I copied it over to XenServer and tried to install it I got a `cpio: Bad magic` error. So, for testing for now, I've manually copied over the neutron_vif_flow to /etc/xapi.d/plugins, ran chmod a+x, and invoked it from a XenAPI session from the compute node. I learned that plugins can only take in strings and must return strings out. I've updated the PR as such.

**Comment by asadoughi** [comment]
Removed WIP from title. Please review.

File: quark/agent/tests/test_xapi.py [hunk 1]
```diff
@@ -43,17 +41,26 @@ def test_get_instances(self):
...
+                        "domid": "1"},
             "opaque2": {"other_config": {},
                         "power_state": "", "is_a_template": True,
                         "is_control_domain": True,
-                        "name_label": ""},
+                        "name_label": "",
+                        "VIFs": [],
+                        "domid": "2"},
         }
         instances = self.xclient.get_instances()
```
Comment (insequent) [inline quark/agent/tests/test_xapi.py]: So, I kinda wish that this was more clear about only showing power_state=running instances. Just a side note though, nothing to do with this PR.

File: quark/agent/plugins/xenapi/etc/xapi.d/plugins/neutron_vif_flow [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def online_instance_flows(session, arg_dict):
+    domid = arg_dict['dom_id']
+    vif_index = arg_dict['vif_index']
+
+    command = script_command % (domid, vif_index)
+    retcode = get_return_code(*command.split(' '))
+    if retcode:
+        raise Exception('online_instance_flows command failed |%s|' % command)
```
Comment (insequent) [inline quark/agent/plugins/xenapi/etc/xapi.d/plugins/neutron_vif_flow]: Might want to grab stderr no? Or even combine stdout and stderr, and return that?

**Comment by insequent** [comment]
LGTM, besides little notes above.

Since the rpm is literally a file drop (i.e. no real compiling), fixing the rpmbuild for XenStore should be relatively easy, no? Want to hash that out first?

File: quark/agent/plugins/xenapi/etc/xapi.d/plugins/neutron_vif_flow (hunk 2 above)
Comment (asadoughi) [inline quark/agent/plugins/xenapi/etc/xapi.d/plugins/neutron_vif_flow]: This script, I copied from an existing nova plugin so I didn't think about it much. I'm not sure what the limits are on the length of the string returned from here.

**Comment by asadoughi** [comment]
I think the RPM is fine. I'm just not sure what went wrong between Fedora VM and trying to install it in XenServer. I think I'd like to get this merged and debug it once everything is integrated if it fails, jenkins will be creating the RPM .

File: quark/agent/plugins/xenapi/etc/xapi.d/plugins/neutron_vif_flow (hunk 2 above)
Comment (andyhky) [inline quark/agent/plugins/xenapi/etc/xapi.d/plugins/neutron_vif_flow]: Just the return code would be plenty in this case. The stderr/stdout is already going to be logged locally on the hypervisor.

**Comment by insequent** [comment]
LGTM :snowman:

**Comment by jmeridth** [comment]
:+1:

**Pull request merged**